Microsoft.HealthVault.Fhir/Transformers/ObservationToHealthVault.cs

```diff
@@ -68,22 +71,37 @@ public static ThingBase ToHealthVault(this Observation observation)
 
         internal static T GetThingValueFromQuantity<T>(Quantity value) where T : Measurement<double>, new()
         {
-            // TODO: detect the units from the code (value.Unit)
-            if (value != null)
+            if (value?.Value == null)
             {
-                if (value.Value.HasValue)
-                {
-                    var result = new T();
-                    result.Value = (double)value.Value;
-                    result.DisplayValue = new DisplayValue();
-                    result.DisplayValue.Value = (double)value.Value;
-                    result.DisplayValue.Units = value.Unit;
-                    result.DisplayValue.UnitsCode = value.Code;
-                    return result;
-                }
+                return null;
             }
+            
+            var unitConversion = UnitResolver.Instance.UnitConversions.FirstOrDefault(x => x.Code == value.Code);
```

Comment:
code is a string ? .. if so .. is it better to use OrdinalIgnoreCase while comparing ?